core/src/executor/evaluate/mod.rs

```diff
@@ -48,6 +49,72 @@ pub async fn evaluate_stateless<'a, 'b: 'a>(
     evaluate_inner(storage, context, None, expr).await
 }
 
+async fn stream_contains_target<'life, 'stream, 'item>(
+    target: Evaluated<'life>,
+    negated: bool,
+    stream: impl Stream<Item = Result<Evaluated<'item>>> + 'stream,
+) -> Result<Nullable<bool>> {
```

Comment:
I believe we can handle this part more simply. The key issue is the negate evaluation, but I haven't fully figured out how to approach it yet. I'll look into it further and leave a comment once I have a better idea.

Reply:
I have unified the branches of the method in commit [b0eaeff](https://github.com/gluesql/gluesql/pull/1577/commits/b0eaeffe300009f8e604e9560d2b74be6752e7b8). While this makes it less code, I am worried it severely reduces its readability. Let me know which one you prefer.